images/bridge2/cmd/bridge/main.go

```diff
@@ -351,6 +354,42 @@ var upgrader = websocket.Upgrader{
 	CheckOrigin: func(r *http.Request) bool { return true },
 }
 
+func (m *Main) serveSessionText(ctx context.Context, w http.ResponseWriter, r *http.Request, sess *Session) {
+	snapshot := sess.Snapshot()
+	content, err := fs.ReadFile(ui.Dir, "session.tmpl.txt")
+	if err != nil {
+		log.Printf("error reading template for session %s: %s", snapshot.ID, err)
+		http.Error(w, "template read error", http.StatusInternalServerError)
+		return
+	}
+	tmpl, err := template.New("session.tmpl.txt").
+		Funcs(template.FuncMap{
+			"json": func(v any) (string, error) {
+				b, err := json.Marshal(v)
+				return string(b), err
+			},
+			"join":   strings.Join,
+			"string": func(v any) string { return v.(fmt.Stringer).String() },
+			"duration": func(s string) (time.Duration, error) {
+				return time.ParseDuration(s)
+			},
+		}).
+		Parse(string(content))
```

Comment:
I think you can use ParseFS here instead of separately opening the file above:
```suggestion
		ParseFS(ui.Dir, "session.tmpl.txt")
```